bokehjs/src/lib/models/widgets/tables/data_table.ts

```diff
@@ -353,8 +418,12 @@ export class DataTable extends TableWidget {
     this.prototype.default_view = DataTableView
 
     this.define<DataTable.Props>({
+      autosize_mode:       [ p.Any, "force_fit" ],
+      auto_edit:           [ p.Boolean, false ],
       columns:             [ p.Array,   []    ],
-      fit_columns:         [ p.Boolean, true  ],
+      fit_columns:         [ p.Boolean, null  ],
+      frozen_columns:      [ p.Int,     null  ],
+      frozen_rows:         [ p.Int,     null  ],
```

Comment:
Given that I already merged PR #10185, there's a new API for defining properties to play with, which I suggest we start slowly migrating to. You can see `LayoutDOM` and `AbstractSlider` as early migrated models.

Reply:
Doesn't seem happy:

    [17:42:08] failed: There were 1 TypeScript errors:
    src/lib/models/widgets/tables/data_table.ts:421:34 - error TS2345: Argument of type '({ Any, Array, Boolean, Int, String }: typeof import("/Users/philippjfr/development/bokeh/bokehjs/src/lib/core/kinds")) => { autosize_mode: [Any, "force_fit"]; auto_edit: [Boolean, false]; columns: [<ItemType>(item_type: Kind<ItemType>) => Array<ItemType>, never[]]; fit_columns: [Boolean]; frozen_columns: [Int]; frozen_rows: [Int]; sortable: [Boolean, true]; reorderable: [Boolean, true]; editable: [Boolean, false]; selectable: [Any, true]; index_position: [Int, number]; index_header: [String, string]; index_width: [Int, number]; scroll_to_selection: [Boolean, true]; header_row: [Boolean, true]; row_height: [Int, number]; }' is not assignable to parameter of type 'Partial<DefineOf<Props>> | ((types: typeof import("/Users/philippjfr/development/bokeh/bokehjs/src/lib/core/kinds")) => Partial<DefineOf<Props>>)'.
      Type '({ Any, Array, Boolean, Int, String }: typeof import("/Users/philippjfr/development/bokeh/bokehjs/src/lib/core/kinds")) => { autosize_mode: [Any, "force_fit"]; auto_edit: [Boolean, false]; columns: [<ItemType>(item_type: Kind<ItemType>) => Array<ItemType>, never[]]; fit_columns: [Boolean]; frozen_columns: [Int]; frozen_rows: [Int]; sortable: [Boolean, true]; reorderable: [Boolean, true]; editable: [Boolean, false]; selectable: [Any, true]; index_position: [Int, number]; index_header: [String, string]; index_width: [Int, number]; scroll_to_selection: [Boolean, true]; header_row: [Boolean, true]; row_height: [Int, number]; }' is not assignable to type '(types: typeof import("/Users/philippjfr/development/bokeh/bokehjs/src/lib/core/kinds")) => Partial<DefineOf<Props>>'.
        Call signature return types '{ autosize_mode: [Any, "force_fit"]; auto_edit: [Boolean, false]; columns: [<ItemType>(item_type: Kind<ItemType>) => Array<ItemType>, never[]]; fit_columns: [Boolean]; frozen_columns: [Int]; frozen_rows: [Int]; sortable: [Boolean, true]; reorderable: [Boolean, true]; editable: [Boolean, false]; selectable: [Any, true]; index_position: [Int, number]; index_header: [String, string]; index_width: [Int, number]; scroll_to_selection: [Boolean, true]; header_row: [Boolean, true]; row_height: [Int, number]; }' and 'Partial<DefineOf<Props>>' are incompatible.
          The types of 'columns[0]' are incompatible between these types.
            Type '<ItemType>(item_type: Kind<ItemType>) => Kinds.Array<ItemType>' is not assignable to type 'PropertyConstructor<TableColumn[]> | Kind<TableColumn[]>'.
              Type '<ItemType>(item_type: Kind<ItemType>) => Array<ItemType>' is missing the following properties from type 'Kind<TableColumn[]>': __type__, valid

    421     this.define<DataTable.Props>(({Any, Array, Boolean, Int, String}) => {
                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reply:
I'm guessing you wrote `Array`, instead of `Array(SomeType)`, e.g. `Array(Ref(TableColumn))`.